sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -236,6 +236,41 @@ public Object get(int ordinal) {
   }
 
   @Override
+  public Object get(int ordinal, DataType dataType) {
+    if (dataType instanceof NullType) {
```

Comment:
I found a bug here: we should also check whether the row `isNullAt(ordinal)`, in which case we should return `null`.